code/controllers/subsystem/reagents.dm

```diff
@@ -0,0 +1,73 @@
+SUBSYSTEM_DEF(reagents)
+	name = "Reagents"
+	init_order = SS_INIT_REAGENTS
+	flags = SS_NO_FIRE
+
+/datum/controller/subsystem/reagents/Initialize()
+	// Initalize to create the global chemistry lists:
+	// Must be before SSatoms.InitializeAtoms and SSmapping
+	prepare_properties()
+	prepare_reagents()
+
+	return SS_INIT_SUCCESS
+
+/datum/controller/subsystem/reagents/Recover()
+	initialized = SSreagents.initialized
+
+/datum/controller/subsystem/reagents/proc/prepare_properties()
+	//Chemical Properties - Initialises all /datum/chem_property into a list indexed by property name
+	var/paths = typesof(/datum/chem_property)
+	chemical_properties_list = list()
+	//Some filters
+	chemical_properties_list["negative"] = list()
+	chemical_properties_list["neutral"] = list()
+	chemical_properties_list["positive"] = list()
+	chemical_properties_list["rare"] = list()
+	//Save
+	for(var/path in paths)
+		var/datum/chem_property/prop = new path()
+		if(!prop.name)
+			continue
+		chemical_properties_list[prop.name] = prop
+		if(prop.starter)
+			//Add a separate instance to the chemical property database
+			var/datum/chem_property/chem = new path()
+			chem.level = 0
+			chemical_data.research_property_data += chem
+		if(prop.rarity > PROPERTY_DISABLED)
+			//Filters for the generator picking properties
+			if(prop.rarity == PROPERTY_RARE || prop.rarity == PROPERTY_LEGENDARY)
+				chemical_properties_list["rare"][prop.name] = prop
+			else if(isNegativeProperty(prop))
+				chemical_properties_list["negative"][prop.name] = prop
+			else if(isNeutralProperty(prop))
+				chemical_properties_list["neutral"][prop.name] = prop
+			else if(isPositiveProperty(prop))
+				chemical_properties_list["positive"][prop.name] = prop
+
+/datum/controller/subsystem/reagents/proc/prepare_reagents()
+	//I dislike having these here but map-objects are initialised before world/New() is called. >_>
+	set waitfor = 0
```

Comment:
```suggestion
	set waitfor = FALSE
```

Reply:
TBH I don't know if that set is still even necessary or needed, but the original proc had it and I didn't notice any issues.